heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/MetricsManager.java

```diff
@@ -328,24 +332,32 @@ public void start() {
      * Handler for uncaughtException
      */
     public void uncaughtException(Thread thread, Throwable exception) {
-      String threadName = thread.getName();
       LOG.log(Level.SEVERE,
-          "Exception caught in thread: " + threadName + " with thread id: " + thread.getId(),
+          "Exception caught in thread: " + thread.getName() + " with thread id: " + thread.getId(),
           exception);
 
-      String sinkId = threadName;
-      Integer thisSinkRetryAttempts = sinksRetryAttempts.remove(sinkId);
+      if (exception instanceof Error || thread.getId() == mainThreadId) {
+        LOG.severe("Would not recover from error. Metrics Manager halts immediately");
```

Comment:
Instead of logging the exception separately above, can we have an if/then with a different log message in each. That way it's clear what's causing what.